dst.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2020 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository at
+// https://github.com/direct-state-transfer/dst-go
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dst defines domain types and interfaces used in this project.
+package dst
+
+import (
+	"context"
+
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/channel/persistence"
+	"perun.network/go-perun/client"
+	perunLog "perun.network/go-perun/log"
+	"perun.network/go-perun/peer"
+	"perun.network/go-perun/wallet"
+)
+
+// Peer represents any participant in the off-chain network that the user wants to transact with.
+type Peer struct {
+	// Name assigned by user for referring to this peer in api requests to the node.
+	// It is unique within a session on the node.
+	Alias string
+
+	OffChainAddr peer.Address // Permanent identity used for authenticating the peer in the off-chain network.
+
+	CommAddr string // Address for off-chain communication
+	CommType string // Type of off-chain communication protocol
+}
+
+//go:generate mockery -name CommBackend -output ./internal/mocks
+
+// CommBackend defines the set of methods required for initializing components required for off-chain communication.
+type CommBackend interface {
+	// Returns a listener that can listen incommig messages at the specified address using the communication protocol.
+	NewListener(address string) (peer.Listener, error)
+
+	// Returns a dialer that can dial for new outgoing connections.
+	// If timeout is zero, program will use no timeout, but standard OS timeouts may still apply.
+	NewDialer() peer.Dialer
+}
+
+// Credential represents the parameters required for a to create a signature for a given address.
+type Credential struct {
+	Addr     wallet.Address
+	Wallet   wallet.Wallet
+	Keystore string
+	Password string
+}
+
+// User represents a participant in the off-chain network that uses a session on this node for sending transactions.
+type User struct {
+	Peer
+
+	OnChain  Credential // Account for funding the channel and the on-chain transactions.
+	OffChain Credential // Account (corresponding to off-chain address) used for signing authentication messages.
+
+	// List of participant addresses for this user in each open channel.
+	// OffChain credential is used for managing all these accounts.
+	PartAddrs []wallet.Address
```

Comment:
The first implementation would use same address for all channels, quickly this will be followed by the update to use ephemeral keys for each channels. This field such ephemeral addresses. The keys corresponding to those addresses can be accessed using the same credentials as the off-chain wallet.